llvm/lib/Target/ARM/Thumb1InstrInfo.cpp

```diff
@@ -135,14 +135,14 @@ void Thumb1InstrInfo::loadRegFromStackSlot(MachineBasicBlock &MBB,
 void Thumb1InstrInfo::expandLoadStackGuard(
     MachineBasicBlock::iterator MI) const {
   MachineFunction &MF = *MI->getParent()->getParent();
-  const TargetMachine &TM = MF.getTarget();
   const ARMSubtarget &ST = MF.getSubtarget<ARMSubtarget>();
+  const auto *GV = cast<GlobalValue>((*MI->memoperands_begin())->getValue());
 
   assert(MF.getFunction().getParent()->getStackProtectorGuard() != "tls" &&
          "TLS stack protector not supported for Thumb1 targets");
 
   unsigned Instr;
-  if (TM.isPositionIndependent())
+  if (!GV->isDSOLocal())
```

Comment:
does Thumb1 need similar guards for whether the target is elf?

Reply:
Based on my archaeology to the  `Thumb1InstrInfo::expandLoadStackGuard` function, the changes haven't been related to Thumb1-only microarchitectures (before v6T2/v7) for non-ELF object file formats.

https://reviews.llvm.org/D75453 says that Windows AArch32 requires at least v7.